### PR TITLE
COVID Banner

### PR DESCRIPTION
**Pull request by kewitz**
## Preview
COVID_desktop

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "@apollo/react-ssr": "^3.1.3",
     "@formatjs/intl-relativetimeformat": "4.5.10",
     "@material-ui/core": "4.9.7",
+    "@n8tb1t/use-scroll-position": "^1.0.43",
```
Comment (znarf) [inline package.json]: What is that package, could it be avoided ?

Comment (kewitz) [reply package.json]: It is a small react-hook for scrolling down effects, it is processing the logic related to docking the banner over the footer.
I can rewrite that using `useEffect` but we'll be able to deprecate that with the banner when we don't use it anymore.

File: components/DismissibleMessage.js [hunk 2]
```diff
@@ -43,23 +45,32 @@ const DismissibleMessage = ({ LoggedInUser, messageId, children }) => {
...
     return null;
   }
 
   return children({
     dismiss: () => {
       setDismissedLocally(true);
-      return editAccountSettings({
-        variables: { account: { id: loggedInAccount.id }, key: settingsKey },
-      });
+      window?.localStorage.setItem(settingsKey, 'true');
```
Comment (Betree) [inline components/DismissibleMessage.js]: Check `lib/local-storage.js` it has helpers to do that safely & cross-browser

```suggestion
      setLocalStorage(settingsKey, 'true');
```

File: components/DismissibleMessage.js [hunk 3]
```diff
@@ -30,9 +30,11 @@ const editAccountSettingsMutation = gqlV2`
  *
  * Messages will never be displayed if user is not logged in.
  */
-const DismissibleMessage = ({ LoggedInUser, messageId, children }) => {
+const DismissibleMessage = ({ LoggedInUser, messageId, displayForLoggedOutUser, children }) => {
   const settingsKey = `${DISMISSABLE_HELP_MESSAGE_KEY}.${messageId}`;
-  const [isDismissedLocally, setDismissedLocally] = React.useState(false);
+  const [isDismissedLocally, setDismissedLocally] = React.useState(
+    typeof window !== 'undefined' && window.localStorage.getItem(settingsKey) === 'true',
```
Comment (Betree) [inline components/DismissibleMessage.js]: Check `lib/local-storage.js` it has helpers to do that safely & cross-browser

```suggestion
      getFromLocalStorage(settingsKey) === 'true';
```

File: package.json (hunk 1 above)
Comment (Betree) [reply package.json]: You can try using css `position: sticky`, it's what we use for the new collective page and in other places. The pure-css approach has better performances and a good browser support nowadays.

File: components/DismissibleMessage.js (hunk 3 above)
Comment (kewitz) [reply components/DismissibleMessage.js]: Nice! Thanks!

File: components/DismissibleMessage.js (hunk 2 above)
Comment (kewitz) [reply components/DismissibleMessage.js]: Thanks!

File: package.json (hunk 1 above)
Comment (Betree) [reply package.json]: A few tips if you go with that (I struggled with them):
- Make sure the parent has `position: relative`
- Make sure to set a `top` (or `bottom` based on the behavior you want)

Comment (kewitz) [reply package.json]: Nice, I'll try that.

**Comment by znarf** [comment]
I would not display the banner on SSR to avoid a flash (displayed / hidden) when loading the page and already dismissed.

When dismissed, how long is it?

**Comment by kewitz** [comment]
Hi @znarf, I disabled the SSR for the banner. I also had to fix the sticky position.

**Pull request merged**